src/clr-angular/forms/select/providers/option-selection.service.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {Injectable} from "@angular/core";
+import {Observable, Subject} from "rxjs";
+
+import {ClrOption} from "../option";
+
+@Injectable()
+export class OptionSelectionService {
+    private currentSelection: ClrOption;
```

Comment:
I'm sensing a refactor as soon as we hit multi-selection. But I'd rather not have the same mess as datagrid...

Maybe always having an array, and single selection will always be an array of 0 or 1 option?

Reply:
I spoke about this to you over chat. We decided to keep it this way and change things when we implement Multiple Selection